group-coordinator/src/test/java/org/apache/kafka/coordinator/group/GroupMetadataManagerTest.java

```diff
@@ -1807,9 +1776,9 @@ public void testReconciliationProcess() {
 
         assertRecordsEquals(Collections.singletonList(
             RecordHelpers.newCurrentAssignmentRecord(groupId, new ConsumerGroupMember.Builder(memberId1)
+                .setState(MemberState.UNREVOKED_PARTITIONS)
                 .setMemberEpoch(10)
-                .setPreviousMemberEpoch(9)
-                .setTargetMemberEpoch(11)
+                .setPreviousMemberEpoch(10)
```

Comment:
this was bumped from 9 to 10. this seems right because Member 1 was at epoch 10 previously at L1656. how did the test past before?

Reply:
In the [previous implementation](https://github.com/apache/kafka/pull/15364/files#diff-6c5e23803064f4b7a122eee29736d036b9cfe244e69f48751ba163d62e2bf35fL257), the previous epoch was not updated. It feels like a bug. In the new implementation, I have set the previous epoch whenever the member epoch is updated.